### ISSUE TITLE
Panic in validator/src/tendermint.rs: Option::unwrap() on a None value

### ISSUE THREAD

**Issue by fiaxh** [nimiq/core-rs-albatross#343]
Happens reproducibly when trying to create block #127.

<details><summary>Full trace log</summary>

```
seed3_1    |  2021-09-22 10:59:48.807 INFO  micro                | [3] Our turn at #127:0, producing micro block
seed3_1    |  2021-09-22 10:59:51.954 ERROR panic                | thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value': /home/fiaxh/Devel/core-rs/validator/src/tendermint.rs:354
seed3_1    |    0: log_panics::init::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/log-panics-2.0.0/src/lib.rs:52:25
seed3_1    |    1: std::panicking::rust_panic_with_hook
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:628:17
seed3_1    |    2: std::panicking::begin_panic_handler::{{closure}}
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:519:13
seed3_1    |    3: std::sys_common::backtrace::__rust_end_short_backtrace
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/sys_common/backtrace.rs:141:18
seed3_1    |    4: rust_begin_unwind
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:517:5
seed3_1    |    5: core::panicking::panic_fmt
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/panicking.rs:96:14
seed3_1    |    6: core::panicking::panic
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/panicking.rs:50:5
seed3_1    |    7: core::option::Option<T>::unwrap
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/option.rs:735:21
seed3_1    |    8: <nimiq_validator::tendermint::TendermintInterface<N> as nimiq_tendermint::outside_deps::TendermintOutsideDeps>::broadcast_and_aggregate::{{closure}}
seed3_1    |              at /home/fiaxh/Devel/core-rs/validator/src/tendermint.rs:354:44
seed3_1    |    9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/future/mod.rs:80:19
seed3_1    |   10: <core::pin::Pin<P> as core::future::future::Future>::poll
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/future/future.rs:119:9
seed3_1    |   11: nimiq_tendermint::network::<impl nimiq_tendermint::tendermint::Tendermint<ProposalTy,ProofTy,ResultTy,DepsTy>>::broadcast_and_aggregate_prevote::{{closure}}
seed3_1    |              at /home/fiaxh/Devel/core-rs/tendermint/src/network.rs:84:27
seed3_1    |   12: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/future/mod.rs:80:19
seed3_1    |   13: nimiq_tendermint::protocol::<impl nimiq_tendermint::tendermint::Tendermint<ProposalTy,ProofTy,ResultTy,DepsTy>>::on_timeout_propose::{{closure}}
seed3_1    |              at /home/fiaxh/Devel/core-rs/tendermint/src/protocol.rs:181:9
seed3_1    |   14: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/future/mod.rs:80:19
seed3_1    |   15: nimiq_tendermint::stream::expect_block::{{closure}}
seed3_1    |              at /home/fiaxh/Devel/core-rs/tendermint/src/stream.rs:73:45
seed3_1    |   16: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/future/mod.rs:80:19
seed3_1    |   17: <async_stream::async_stream::AsyncStream<T,U> as futures_core::stream::Stream>::poll_next
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/async-stream-0.3.2/src/async_stream.rs:53:17
seed3_1    |   18: <futures_util::stream::stream::map::Map<St,F> as futures_core::stream::Stream>::poll_next
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.17/src/stream/stream/map.rs:58:26
seed3_1    |   19: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.17/src/stream.rs:120:9
seed3_1    |   20: futures_util::stream::stream::StreamExt::poll_next_unpin
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.17/src/stream/stream/mod.rs:1474:9
seed3_1    |   21: <futures_util::stream::stream::into_future::StreamFuture<St> as core::future::future::Future>::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.17/src/stream/stream/into_future.rs:85:20
seed3_1    |   22: <futures_util::stream::futures_unordered::FuturesUnordered<Fut> as futures_core::stream::Stream>::poll_next
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.17/src/stream/futures_unordered/mod.rs:521:17
seed3_1    |   23: futures_util::stream::stream::StreamExt::poll_next_unpin
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.17/src/stream/stream/mod.rs:1474:9
seed3_1    |   24: <futures_util::stream::select_all::SelectAll<St> as futures_core::stream::Stream>::poll_next
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.17/src/stream/select_all.rs:100:26
seed3_1    |   25: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.17/src/stream.rs:120:9
seed3_1    |   26: futures_util::stream::stream::StreamExt::poll_next_unpin
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.17/src/stream/stream/mod.rs:1474:9
seed3_1    |   27: <nimiq_tendermint::stream::TendermintStreamWrapper<DepsTy,ProposalTy,ProofTy,ResultTy> as futures_core::stream::Stream>::poll_next
seed3_1    |              at /home/fiaxh/Devel/core-rs/tendermint/src/stream.rs:157:15
seed3_1    |   28: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-core-0.3.17/src/stream.rs:120:9
seed3_1    |   29: futures_util::stream::stream::StreamExt::poll_next_unpin
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.17/src/stream/stream/mod.rs:1474:9
seed3_1    |   30: <nimiq_validator::macro::ProduceMacroBlock as futures_core::stream::Stream>::poll_next
seed3_1    |              at /home/fiaxh/Devel/core-rs/validator/src/macro.rs:131:9
seed3_1    |   31: futures_util::stream::stream::StreamExt::poll_next_unpin
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.17/src/stream/stream/mod.rs:1474:9
seed3_1    |   32: nimiq_validator::validator::Validator<TNetwork,TValidatorNetwork>::poll_macro
seed3_1    |              at /home/fiaxh/Devel/core-rs/validator/src/validator.rs:342:46
seed3_1    |   33: <nimiq_validator::validator::Validator<TNetwork,TValidatorNetwork> as core::future::future::Future>::poll
seed3_1    |              at /home/fiaxh/Devel/core-rs/validator/src/validator.rs:497:17
seed3_1    |   34: <core::pin::Pin<P> as core::future::future::Future>::poll
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/future/future.rs:119:9
seed3_1    |   35: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/core.rs:161:17
seed3_1    |   36: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/loom/std/unsafe_cell.rs:14:9
seed3_1    |   37: tokio::runtime::task::core::CoreStage<T>::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/core.rs:151:13
seed3_1    |   38: tokio::runtime::task::harness::poll_future::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/harness.rs:461:19
seed3_1    |   39: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/panic/unwind_safe.rs:271:9
seed3_1    |   40: std::panicking::try::do_call
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:403:40
seed3_1    |   41: __rust_try
seed3_1    |   42: std::panicking::try
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:367:19
seed3_1    |   43: std::panic::catch_unwind
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panic.rs:129:14
seed3_1    |   44: tokio::runtime::task::harness::poll_future
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/harness.rs:449:18
seed3_1    |   45: tokio::runtime::task::harness::Harness<T,S>::poll_inner
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/harness.rs:98:27
seed3_1    |   46: tokio::runtime::task::harness::Harness<T,S>::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/harness.rs:53:15
seed3_1    |   47: tokio::runtime::task::raw::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/raw.rs:113:5
seed3_1    |   48: tokio::runtime::task::raw::RawTask::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/raw.rs:70:18
seed3_1    |   49: tokio::runtime::task::LocalNotified<S>::run
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/mod.rs:343:9
seed3_1    |   50: tokio::runtime::thread_pool::worker::Context::run_task::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/thread_pool/worker.rs:408:13
seed3_1    |   51: tokio::coop::with_budget::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/coop.rs:106:9
seed3_1    |   52: std::thread::local::LocalKey<T>::try_with
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/thread/local.rs:399:16
seed3_1    |   53: std::thread::local::LocalKey<T>::with
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/thread/local.rs:375:9
seed3_1    |   54: tokio::coop::with_budget
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/coop.rs:99:5
seed3_1    |       tokio::coop::budget
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/coop.rs:76:5
seed3_1    |       tokio::runtime::thread_pool::worker::Context::run_task
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/thread_pool/worker.rs:407:9
seed3_1    |   55: tokio::runtime::thread_pool::worker::Context::run
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/thread_pool/worker.rs:374:24
seed3_1    |   56: tokio::runtime::thread_pool::worker::run::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/thread_pool/worker.rs:359:17
seed3_1    |   57: tokio::macros::scoped_tls::ScopedKey<T>::set
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/macros/scoped_tls.rs:61:9
seed3_1    |   58: tokio::runtime::thread_pool::worker::run
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/thread_pool/worker.rs:356:5
seed3_1    |   59: tokio::runtime::thread_pool::worker::Launch::launch::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/thread_pool/worker.rs:335:45
seed3_1    |   60: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/blocking/task.rs:42:21
seed3_1    |   61: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/core.rs:161:17
seed3_1    |   62: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/loom/std/unsafe_cell.rs:14:9
seed3_1    |   63: tokio::runtime::task::core::CoreStage<T>::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/core.rs:151:13
seed3_1    |   64: tokio::runtime::task::harness::poll_future::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/harness.rs:461:19
seed3_1    |   65: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/panic/unwind_safe.rs:271:9
seed3_1    |   66: std::panicking::try::do_call
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:403:40
seed3_1    |   67: __rust_try
seed3_1    |   68: std::panicking::try
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:367:19
seed3_1    |   69: std::panic::catch_unwind
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panic.rs:129:14
seed3_1    |   70: tokio::runtime::task::harness::poll_future
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/harness.rs:449:18
seed3_1    |   71: tokio::runtime::task::harness::Harness<T,S>::poll_inner
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/harness.rs:98:27
seed3_1    |   72: tokio::runtime::task::harness::Harness<T,S>::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/harness.rs:53:15
seed3_1    |   73: tokio::runtime::task::raw::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/raw.rs:113:5
seed3_1    |   74: tokio::runtime::task::raw::RawTask::poll
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/raw.rs:70:18
seed3_1    |   75: tokio::runtime::task::UnownedTask<S>::run
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/task/mod.rs:379:9
seed3_1    |   76: tokio::runtime::blocking::pool::Inner::run
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/blocking/pool.rs:265:17
seed3_1    |   77: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
seed3_1    |              at /home/fiaxh/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.11.0/src/runtime/blocking/pool.rs:245:17
seed3_1    |   78: std::sys_common::backtrace::__rust_begin_short_backtrace
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/sys_common/backtrace.rs:125:18
seed3_1    |   79: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/thread/mod.rs:481:17
seed3_1    |   80: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/panic/unwind_safe.rs:271:9
seed3_1    |   81: std::panicking::try::do_call
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:403:40
seed3_1    |   82: __rust_try
seed3_1    |   83: std::panicking::try
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panicking.rs:367:19
seed3_1    |   84: std::panic::catch_unwind
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/panic.rs:129:14
seed3_1    |   85: std::thread::Builder::spawn_unchecked::{{closure}}
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/thread/mod.rs:480:30
seed3_1    |   86: core::ops::function::FnOnce::call_once{{vtable.shim}}
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/core/src/ops/function.rs:227:5
seed3_1    |   87: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/alloc/src/boxed.rs:1636:9
seed3_1    |       <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/alloc/src/boxed.rs:1636:9
seed3_1    |       std::sys::unix::thread::Thread::new::thread_start
seed3_1    |              at /rustc/9bb77da74dac4768489127d21e32db19b59ada5b/library/std/src/sys/unix/thread.rs:106:17
seed3_1    |   88: start_thread
seed3_1    |   89: clone
```
</details>

[Logs of all 4 validators](https://github.com/nimiq/core-rs-albatross/files/7210676/tendermint_unwrap.zip)

**Comment by nibhar** [comment]
My initial assessment is that the `broadcast_and_aggregate` function incorrectly assumes that a proposal was received to extract the pk_tree_root from.

**Comment by fiaxh** [comment]
This issue has been caused by recent changes to `pk_tree_root` and was resolved by reverting them for now (8fd2af5c65491bd94e879800642e31a75aca358c, fab7fff9505dcff5da27843201bb1ccf37a6d188, 8b84dc1e6df7ae77914278adef9fc18f5b8db7c7).
